src/components/ResourceList/tests/ResourceList.test.tsx

```diff
@@ -1197,6 +1198,90 @@ describe('<ResourceList />', () => {
       expect(resourceList.find(BulkActions).prop('selectMode')).toBe(false);
     });
   });
+
+  describe('isFiltered', () => {
+    it('renders `selectAllFilteredItems` label if true', () => {
+      const resourceList = mountWithAppProvider(
+        <ResourceList
+          items={itemsNoID}
+          resourceName={{singular: 'customer', plural: 'customers'}}
+          hasMoreItems
+          renderItem={renderItem}
+          bulkActions={bulkActions}
+          isFiltered
+        />,
+      );
+
+      expect(
+        resourceList.find(BulkActions).prop('paginatedSelectAllAction'),
+      ).toStrictEqual({
+        content: 'Select all 2+ customers in this filter',
+        onAction: expect.any(Function),
+      });
+    });
+
+    it('renders `allFilteredItemsSelected` label if true and all items are selected', () => {
+      const resourceList = mountWithApp(
+        <ResourceList
+          items={itemsNoID}
+          selectedItems={SELECT_ALL_ITEMS}
+          resourceName={{singular: 'customer', plural: 'customers'}}
+          hasMoreItems
+          renderItem={renderItem}
+          bulkActions={bulkActions}
+          selectable
+          isFiltered
+        />,
+      );
+
+      resourceList.find(BulkActions)!.find(Button)!.trigger('onClick');
+
+      expect(
+        resourceList.find(BulkActions)!.prop('paginatedSelectAllText'),
+      ).toBe('All 2+ customers in this filter are selected.');
+    });
+
+    it('renders `selectAllItems` label if false', () => {
```

Comment:
:bike: 🐑 For these tests, we could just not provide it (not explicitly passing in `false` as a prop), and test that default behavior, ie: `renders selectAllItems label if isFiltered prop is not provided`